### PR TITLE
Feature/tls sni info extension

### PR DESCRIPTION
**Pull request by Florian-Limpoeck**

### CODE REVIEW

File: src/main/java/com/hivemq/security/ssl/SslSniHandler.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        if (hostname != null) {
+            ctx.channel().attr(ChannelAttributes.AUTH_SNI_HOSTNAME).set(hostname);
+            if (log.isTraceEnabled()) {
+                log.trace("Client with IP '{}' sent SNI hostname '{}'", ChannelUtils.getChannelIP(ctx.channel()).or("UNKNOWN"), hostname);
+            }
+        }
+
+        SslHandler sslHandlerInstance = null;
+        try {
```
Comment (DC2-DanielKrueger) [inline src/main/java/com/hivemq/security/ssl/SslSniHandler.java:64]: why dont we call the super() of the SniHandler?

File: src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ */
+public abstract class AbstractTlsChannelInitializer extends AbstractChannelInitializer {
+
+    private final @NotNull TlsListener tlsListener;
+    private final @NotNull SslFactory sslFactory;
+    private final @NotNull ChannelDependencies channelDependencies;
+
+    public AbstractTlsChannelInitializer(@NotNull final ChannelDependencies channelDependencies,
+                                         @NotNull final TlsListener tlsListener,
+                                         @NotNull final SslFactory sslFactory) {
```
Comment (gitseti) [inline src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java:49]:
```suggestion
    public AbstractTlsChannelInitializer(final @NotNull ChannelDependencies channelDependencies,
                                         final @NotNull TlsListener tlsListener,
                                         final @NotNull SslFactory sslFactory) {
```

File: src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        super(channelDependencies, tlsListener);
+        this.tlsListener = tlsListener;
+        this.sslFactory = sslFactory;
+        this.channelDependencies = channelDependencies;
+    }
+
+    @Override
+    protected void addNoConnectIdleHandler(@NotNull final Channel ch) {
```
Comment (gitseti) [inline src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java:58]:
```suggestion
    protected void addNoConnectIdleHandler(final @NotNull Channel ch) {
```

File: src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        // No connect idle handler are added, as soon as the TLS handshake is done.
+    }
+
+    protected void addNoConnectIdleHandlerAfterTlsHandshake(@NotNull final Channel ch) {
```
Comment (gitseti) [inline src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java:62]:
```suggestion
    protected void addNoConnectIdleHandlerAfterTlsHandshake(final @NotNull Channel ch) {
```

File: src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        super.addNoConnectIdleHandler(ch);
+    }
+
+    @Override
+    protected void addSpecialHandlers(@NotNull final Channel ch) throws SslException {
```
Comment (gitseti) [inline src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractTlsChannelInitializer.java:67]:
```suggestion
    protected void addSpecialHandlers(final @NotNull Channel ch) throws SslException {
```

File: src/main/java/com/hivemq/extensions/client/parameter/ConnectionInformationImpl.java [hunk 6]
```diff
@@ -37,7 +37,7 @@
     private final @NotNull ConnectionAttributeStore connectionAttributeStore;
     private final @Nullable InetAddress inetAddress;
     private final @Nullable Listener listener;
-    private final @Nullable TlsInformation tlsInformation;
+    private final @Nullable ClientTlsInformation tlsInformation;
```
Comment (gitseti) [inline src/main/java/com/hivemq/extensions/client/parameter/ConnectionInformationImpl.java:40]:
```suggestion
    private final @Nullable ClientTlsInformation clientTlsInformation;
```

**Review by gitseti — APPROVED** [review]

**Review by DC2-DanielKrueger — APPROVED** [review]

File: src/main/java/com/hivemq/security/ssl/SslSniHandler.java (hunk 1 above)
Comment (Florian-Limpoeck) [reply src/main/java/com/hivemq/security/ssl/SslSniHandler.java:64]: because we already have an SSLHandler

**Pull request closed without merging**